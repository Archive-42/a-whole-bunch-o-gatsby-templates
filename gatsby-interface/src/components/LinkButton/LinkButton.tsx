/** @jsx jsx */
import { jsx } from "@emotion/core"
import { BaseLink, BaseLinkProps } from "../BaseLink"
import { getButtonStyles, ButtonStyleProps } from "../Button"

export type LinkButtonProps<TState = any> = BaseLinkProps<TState> &
  ButtonStyleProps

export function LinkButton<TState>(props: LinkButtonProps<TState>) {
  const { children, size, tone, variant, leftIcon, rightIcon, ...rest } = props

  return (
    <BaseLink
      {...getButtonStyles({
        children,
        size,
        tone,
        variant,
        leftIcon,
        rightIcon,
      })}
      {...rest}
    />
  )
}
