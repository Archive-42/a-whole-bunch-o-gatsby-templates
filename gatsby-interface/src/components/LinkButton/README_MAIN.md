## LinkButton

Built on the basis of [`BaseLink`](??path=/story/baselink--tags-components)

```javascript
import { LinkButton } from "gatsby-interface"

function Component() {
  return (
    <LinkButton to="/dashboard">Label</AnchorButton>
  )
}
```

### Props

`LinkButton` accepts all props accepted by Gatsby's [`Link`](https://www.gatsbyjs.org/docs/gatsby-link/) component as well as `ButtonStyleProps`
