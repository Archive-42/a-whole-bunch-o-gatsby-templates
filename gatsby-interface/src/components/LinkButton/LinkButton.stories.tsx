/** @jsx jsx */
import { storiesOf } from "@storybook/react"
import README_MAIN from "./README_MAIN.md"
import README_ICONS from "../Button/README_ICONS.md"
import customStyling from "../Button/README_customStyling.md"
import { LinkButton, LinkButtonProps } from "./"
import {
  showcaseVariants,
  showcaseSizes,
  showcaseTones,
  showcaseCustomStyles,
  showcaseIcons,
} from "../Button/utils/storybook-styles"

const defaultProps: LinkButtonProps = {
  to: `/`,
}

storiesOf(`buttons/LinkButton`, module)
  .addParameters({
    options: {
      showPanel: true,
    },
    readme: {
      sidebar: README_MAIN,
      includePropTables: [LinkButton],
    },
  })
  .add(...showcaseVariants(LinkButton, defaultProps))
  .add(...showcaseSizes(LinkButton, defaultProps))
  .add(...showcaseTones(LinkButton, defaultProps))
  .add(...showcaseCustomStyles(LinkButton, defaultProps, customStyling))
  .add(...showcaseIcons(LinkButton, defaultProps, README_ICONS))
