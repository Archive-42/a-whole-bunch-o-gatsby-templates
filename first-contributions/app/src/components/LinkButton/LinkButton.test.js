import React from 'react';
import renderer from 'react-test-renderer';
import LinkButton from './LinkButton';


it('renders a project card', () => {
  const tree = renderer.create(
    <LinkButton />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

