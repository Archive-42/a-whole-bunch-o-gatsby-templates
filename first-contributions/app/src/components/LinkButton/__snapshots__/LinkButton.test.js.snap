// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a project card 1`] = `
<a
  className="LinkButton"
  href="https://github.com/Roshanjossey/first-contributions/blob/master/README.md"
>
  <span>
     Get started 
  </span>
</a>
`;
